ocaml/xapi/xapi_pool.ml

```diff
@@ -666,9 +673,8 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
            , [Features.name_of_feature Features.Pool_size] ))
   in
   let assert_no_hosts_in_updating () =
-    let pool = Helpers.get_pool ~__context in
     if List.exists (fun (_, op) -> op = `apply_updates)
-        (Client.Pool.get_current_operations ~rpc ~session_id ~self:pool) then
+        (Client.Pool.get_current_operations ~rpc ~session_id ~self:remote_pool) then
```

Comment:
This does indeed look like a bug to me. I thought that maybe we already perform this check somewhere else in the code for the remote pool instead, but I can't find it! The only problem with this change is that we are now not checking if the joining host is updating

My personal preference would be to perform this check on the remote pool itself (for example you could do it in `xapi_pool.ml:initial_auth` perhaps), and then you can leave the code in `pre_join_checks` alone

Reply:
Many of the pre-join checks look at properties of the remote pool using the API. I may not matter much, unless we put the remote pool into a kind of join-receive state such that it would block conflicting operations during the join. But we currently don't have such a mechanism.

I do think that Ben is right about still needing the local check. I also can't find a local operations check elsewhere.

Reply:
I just added a check on local joining host.